c2cgeoportal/views/pdfreport.py

```diff
@@ -0,0 +1,162 @@
+            headers['content-disposition'] = resp['content-disposition']
+        return Response(
+            content, status=resp.status, headers=headers
+        )
+
+    @view_config(route_name='pdfreport', renderer="json")
+    def getreport(self):
+        id = int(self.request.matchdict['id'])
+        layername = self.request.matchdict['layername']
+        if id is None or layername is None:
+            raise HTTPBadRequest
+        featureid = layername + "." + str(id)
+
+        # check user credentials
+        role_id = None if self.request.user is None else \
+            self.request.user.role.id
+
+        # FIXME: support of mapserver groups
+        if layername in get_private_layers() and \
+                layername not in get_protected_layers(role_id):
+            raise HTTPForbidden
+
+        params = {
+            'service': 'WFS',
+            'version': '1.0.0',
+            'request': 'GetFeature',
+            'typeName': layername,
+            'featureid': featureid,
+            'outputFormat': 'geojson'
+        }
+        srs = self.config.get('srs_name')
+        if srs is not None:
+            params['srsName'] = srs
+
+        request_url = self.config['mapserv_url'] + "?" \
+            + "&".join(["%s=%s" % i for i in params.items()])
+
+        baselayer = self.config.get("layers", {}). \
+            get(layername, {}).get('baselayers')
+        if baselayer is None:
+            baselayer = self.config.get('default_baselayers')
+
+        imageformat = self.config.get("layers", {}). \
+            get(layername, {}).get('imageformat')
+        if imageformat is None:
+            imageformat = self.config.get('default_imageformat')
+
+        spec = {
+            "layout": layername,
+            "outputFormat": "pdf",
+            "attributes": {
+                "paramID": id,
+                "map": {
+                    "projection": srs,
+                    "dpi": 254,
+                    "rotation": 0,
+                    "bbox": [0, 0, 1000000, 1000000],
+                    "zoomToFeatures": {
+                        "zoomType": "center",
+                        "layer": "vector",
+                        "minScale": 25000
+                    },
+                    "layers": [{
+                        "type": "vector",
+                        "name": "vector",
+                        "style": {
+                            "version": "2",
+                            "[1 > 0]": {
+                                "fillColor": "red",
+                                "fillOpacity": 0.2,
+                                "symbolizers": [{
+                                    "strokeColor": "red",
+                                    "strokeWidth": 1,
+                                    "type": "point",
+                                    "pointRadius": 10
+                                }]
+                            }
+                        },
```

Comment:
Actually in some projects the map would not be used. So I think that one should be able to pass the whole ``spec`` in the config.